include/ppx/knob.h

```diff
@@ -357,6 +357,95 @@ class KnobDropdown final
     std::vector<T> mChoices;
 };
 
+// KnobConstant is intended for parameters that cannot be adjusted when the application is run
+// They will not be displayed in the UI and cannot have their values changed by the application
+template <typename T>
+class KnobConstant final
+    : public Knob
+{
+public:
+    KnobConstant(const std::string& flagName, T defaultValue)
+        : Knob(flagName)
+    {
+        static_assert(std::is_same_v<T, bool> || std::is_same_v<T, std::string> || std::is_same_v<T, int> || std::is_same_v<T, float> || std::is_same_v<T, std::pair<int, int>>, "KnobConstant must be of type bool, std::string, int, float, or paired int");
```

Comment:
Usually it's better to assert on the property of a type, not what type it is. i.e. concept, well if the clang version in NDK is newer we could actually use proper language feature instead of static assert